crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -375,6 +427,32 @@ impl<'a> From<&'a mut EntityWorldMut<'_>> for EntityMut<'a> {
     }
 }
 
+impl<'a> TryFrom<FilteredEntityMut<'a>> for EntityMut<'a> {
+    type Error = ();
+
+    fn try_from(value: FilteredEntityMut<'a>) -> Result<Self, Self::Error> {
+        if value.access.has_read_all() && value.access.has_write_all() {
```

Comment:
NIT: AFAIK you don't have to check for read all access, since it's impossible to have write_all access without read_all
```suggestion
        if value.access.has_write_all() {
```

Reply:
I'd like to keep this in as a failsafe.

Reply:
Is this guarantee something that we want to inherit from the semantics of `&` and `&mut` in Rust, to keep it straightforward, or is it possible that we might want to have right-only access in the future?